aexpect/remote_door.py

```diff
@@ -0,0 +1,718 @@
+    local objects separately with more manual configuration on the sharing
+    process. This automatic two-way door opening considers simpler cases.
+
+    This method does not rely on any static (template) controls in order to
+    work because the remote door takes care to reach back the local one.
+    """
+    try:
+        remote_object = Pyro4.Proxy("PYRONAME:" + object_name + "@" + host + ":" + str(port))
+        remote_object._pyroBind()
+    except Pyro4.errors.PyroError:
+        if not session:
+            raise
+
+        # if there is no door on the other side, open one
+        remote.scp_to_remote(session.host, session.port, session.username, session.password,
+                             os.path.abspath(__file__), REMOTE_PYTHON_PATH)
+        run_remote_util(session, "remote_door", "share_local_object",
+                        object_name, host=host, port=port, detach=True)
+        output, attempts = "", 10
+        for _ in range(attempts):
+            output = session.get_output()
+            if "ready" in output:
+                break
+            time.sleep(1)
+        else:
+            raise OSError("Local object sharing failed:\n%s", output)
+        logging.debug("Local object sharing output:\n%s", output)
+        logging.getLogger("Pyro4").setLevel(10)
+
+        remote_object = Pyro4.Proxy("PYRONAME:" + object_name + "@" + host + ":" + str(port))
+        remote_object._pyroBind()
+    return remote_object
+
+
+def get_remote_objects(session=None, host="localhost", port=0):
+    """
+    Get all data objects (full python access) executing remotely or share
+    them if they are not available, generating control files along the way.
+
+    :param session: connection session if the object will be shared
+    :type session: RemoteSession or None
+    :param str host: ip address of the local sharing server
+    :param int port: port of the local sharing server
+    :returns: proxy version of the remote objects
+    :rtype: Pyro4.Proxy
+
+    This method does not rely on any static (template) controls in order to
+    work because the remote door takes care to reach back the local one.
+    """
+    Pyro4.config.SERIALIZER = "pickle"
+    Pyro4.config.PICKLE_PROTOCOL_VERSION = 3
+    from Pyro4.utils import flame
+
+    try:
+        remote_objects = flame.connect(host + ":" + str(port))
+        remote_objects._pyroBind()
+    except Pyro4.errors.PyroError:
+        if not session:
+            raise
+
+        # if there is no door on the other side, open one
+        remote.scp_to_remote(session.host, session.port, session.username, session.password,
+                             os.path.abspath(__file__), REMOTE_PYTHON_PATH)
+        run_remote_util(session, "remote_door", "share_local_objects",
+                        wait=True, host=host, port=port, detach=True)
+        for _ in range(10):
+            if "ready" in session.cmd("cat /tmp/control.log"):
```

Comment:
Let's move these hardcoded paths to module level constants, to make it easier to modify and keep track of them.

Reply:
Fully agreed. Fixed with next push.